homeassistant/components/sensor/nilu_air_quality.py

```diff
@@ -0,0 +1,162 @@
+"""
+Sensor for checking the air quality around Norway.
+
+Data delivered by luftkvalitet.info and nilu.no.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nilu_air_quality/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_LATITUDE, CONF_LONGITUDE
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['niluclient==0.1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=30)
+
+CONF_AREAS = 'areas'
+
+AREAS = [
+    'Ålesund',
+    'Zeppelinfjellet',
+    'Tustervatn',
+    'Trondheim',
+    'Tromsø',
+    'Sør-Varanger',
+    'Stavanger',
+    'Sarpsborg',
+    'Sandve',
+    'Prestebakke',
+    'Oslo',
+    'Narvik',
+    'Moss',
+    'Mo i Rana',
+    'Lørenskog',
+    'Lillestrøm',
+    'Lillesand',
+    'Lillehammer',
+    'Kårvatn',
+    'Kristiansand',
+    'Karasjok',
+    'Hurdal',
+    'Harstad',
+    'Hamar',
+    'Halden',
+    'Grenland',
+    'Gjøvik',
+    'Fredrikstad',
+    'Elverum',
+    'Drammen',
+    'Bærum',
+    'Brumunddal',
+    'Bodø',
+    'Birkenes',
+    'Bergen'
+]
+
+POLLUTION_INDEX = [
+    "No data",
+    'Low',
+    'Moderate',
+    'High',
+    "Extremely high"
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Inclusive(CONF_LATITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.longitude,
+    vol.Optional(CONF_AREAS):
+        vol.All(cv.ensure_list, [vol.In(AREAS)]),
+})
```

Comment:
I would like to suggest to take a different approach for the configuration. From 

```yaml
sensor:
  - platform: nilu_air_quality
    areas:
      - Brumunddal
      - Bergen
```

to

```yaml
sensor:
  - platform: nilu_air_quality
    area: bergen
    monitored_conditions:
      - pm10
      - o3
      - ...
```

This would more align with the configuration of the `luftdaten` sensor which is providing similar data.

Also, creating a sensor for every condition would more align with the configuration through the frontend.

Reply:
I'll take the challenge! Moving the PR to WIP and will start poking the `luftdaten` sensor and draw inspiration from there. 
Thanks.

Reply:
I have changed the configuration to only have one area for each sensor configuration entry, and added possibility to configure which data from the stations should be mounted as an sensor. The sensor has gained a lot of capabilities from it, and hopefully give a more rich sensor platform for users. I have now ran it on my local instance for several days and works really well with new configurations. 
Hopefully you agree that it is improved to the better now. Thanks!